src/ansi-c/expr2c.cpp

```diff
@@ -176,6 +176,34 @@ void expr2ct::get_shorthands(const exprt &expr)
       has_collision=!ns_collision[func].insert(sh).second;
     }
 
+    if(!has_collision)
+    {
+      // We could also conflict with a function argument, the code below
+      // finds the function we're in, and checks we don't conflict with
+      // any argument to the function
+      const std::string symbol_str = id2string(symbol_id);
```

Comment:
Is there a reasons this symbol name/symboilt lookup code is not the same as the code in lines 168-173 above?  In fact, can the checks in this block not just be folded into that existing check block?

Reply:
The check in lines 168-173 is seeing whether the symbol we're looking at now has a conflict with a function symbol name, e.g.
```
void f();

void main(int argc)
{
  int f;
  int g;
```
where here `f` is both a local variable in `main` and also the function `f`.

The new code is finding the current function we're in (e.g. inside `main` above) to check for collisions between local symbols and argument symbols (i.e. above `f` and `g` symbols in the `main` function do not collide with `argc`.